File: Daily/19_03_3191/3191.py
class Solution :
    def minOperations(self, nums: list[int]) -> int:
        n = len(nums)
        operations = 0
        temp_nums = list(nums)
        
        for i in range (n - 2):
            if temp_nums[i] == 0:
                operations += 1
                temp_nums[i] ^= 1
                temp_nums[i + 1] ^= 1
                temp_nums[i + 2] ^= 1
                
        if temp_nums[n - 2] == 0 or temp_nums[n - 1] == 0:
            return -1
        
        return operations

File: Daily/MARCH/19_03_3191/3191.java
class Solution {
    public int minOperations(int[] nums) {
        int  n = nums.length;
        int operations = 0;
        int[] tempNums = nums.clone();

        for (int i = 0; i <= n - 3; i++) {
            if (tempNums[i] == 0) {
                operations++;

                tempNums[i] ^= 1;
                tempNums[ i + 1 ] ^= 1;
                tempNums[ i + 2] ^= 1;
            }
        }

        if (tempNums[n - 2] == 0 || tempNums[n - 1] == 0) {
            return -1;
        }

        return operations;

    }

}